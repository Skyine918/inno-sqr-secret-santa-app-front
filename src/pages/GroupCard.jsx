import * as React from 'react';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';
import { Chip } from '@mui/material';
import SantaTextField from "../ui/SantaMemberEmailField";
import {useState} from "react";
import Button from "@mui/material/Button";
import {patchAssignees, patchWishlist} from "../api/events";
import CircularProgress from "@mui/material/CircularProgress";
import ListItem from "@mui/material/ListItem";
import ListItemAvatar from "@mui/material/ListItemAvatar";
import Avatar from "@mui/material/Avatar";
import ListItemText from "@mui/material/ListItemText";
import List from "@mui/material/List";

function getStatusColor(status) {
    switch (status) {
        case "pending":
            return "info"
        case "creator":
            return "info"
        case "accepted":
            return "success"
        case "denied":
            return "warning"
        default:
            return "primary"
    }
}

export default function GroupCard(props) {
    const [wishlistIsLoading, setWishlistIsLoading] = useState(false);
    const [assignmentIsLoading, setAssignmentIsLoading] = useState(false);
    const [wishlist, setWishlist] = useState("");
    const [wishlistError, setWishlistError] = useState(null);

    const onClickPatchWishlist = async () => {
        setWishlistIsLoading(true)
        patchWishlist(props.user, wishlist, props.event_id)
            .then((data) => {

            })
            .catch(error => {
                if (error.response) {
                    if (error.response.status === 400) {
                        alert(error.response.data.message)
                    }
                }
                setWishlistError(error.toString())
            })
            .finally(() => {
                setWishlistIsLoading(false)
            })
    }

    const onClickAssignGiftees = async () => {
        setAssignmentIsLoading(true)
        patchAssignees(props.user, props.event_id)
            .then((data) => {

            })
            .catch(error => {
                if (error.response) {
                    if (error.response.status === 400) {
                        alert(error.response.data.message)
                    }
                }
                setWishlistError(error.toString())
            })
            .finally(() => {
                setAssignmentIsLoading(false)
            })
    }

    return (
        <Card className="group-card" sx={{ minWidth: 275, marginBottom: "1em" }}>
            <CardContent>
                <Typography variant="h5" component="div">
                    {props.name} <Chip label={props.status} color={getStatusColor(props.status)} />
                </Typography>
                <List sx={{ width: '100%', maxWidth: 360 }}>
                    <ListItem alignItems="flex-start">
                        <ListItemAvatar>
                            <Avatar alt={props.creator.toUpperCase()} src="/static/images/avatar/1.jpg" />
                        </ListItemAvatar>
                        <ListItemText
                            primary={props.creator}
                            secondary={
                                <React.Fragment>
                                    <Typography
                                        sx={{ display: 'inline' }}
                                        component="span"
                                        variant="body2"
                                        color="text.primary"
                                    >
                                    </Typography>
                                    {" Group creator"}
                                </React.Fragment>
                            }
                        />
                    </ListItem>
                </List>
                <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
                    Gift date: {props.date}
                </Typography>
                <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
                    Event holds in {props.location}
                </Typography>
                <Typography sx={{ fontSize: 14 }} color="text.secondary">
                    Total members invited: <b>{props.totalInvitedUsers}</b>
                </Typography>
                <Typography sx={{ fontSize: 14 }} color="text.secondary">
                    Total members Accepted: <b>{props.totalAcceptedUsers}</b>
                </Typography>

                <SantaTextField
                    margin="normal"
                    required
                    fullWidth
                    id="santa-wishlist"
                    type="text"
                    label="Own Wishlist"
                    name="Own Wishlist"
                    defaultValue={wishlist === "" ? props.wishlist : wishlist}
                    autoComplete="santa-event-own-wishlist"
                    autoFocus
                    onChange={(e) => {setWishlist(e.target.value)}}
                    helperText={!!wishlistError ? wishlistError : ''}
                    error={!!wishlistError}
                    multiline
                    rows={4}
                />

                {(wishlistIsLoading || assignmentIsLoading)
                    ? <div style={{display: "flex", justifyContent: "center"}}><CircularProgress color="success"/></div>
                    : <div/>}

                <Button disabled={wishlistIsLoading} variant="contained" color="success" onClick={onClickPatchWishlist}>
                    Edit own wishlist
                </Button>

                {!!props.assignee_wishlist ? <SantaTextField
                    disabled
                    margin="normal"
                    required
                    fullWidth
                    id="santa-wishlist"
                    type="text"
                    label={props.assignee_email + " Wishlist"}
                    name="Assignee Wishlist"
                    defaultValue={props.assignee_wishlist}
                    autoComplete="santa-event-assignee-wishlist"
                    autoFocus
                    onChange={(e) => {setWishlist(e.target.value)}}
                    helperText={!!wishlistError ? wishlistError : "You are assigned to be secret santa of this person. Here is his wishlist"}
                    error={!!wishlistError}
                    multiline
                    rows={4}
                /> : <div></div>}

                <Button disabled={props.creator !== props.user.email} variant="contained" color="warning" onClick={onClickAssignGiftees}>
                    {(props.creator === props.user.email) ? "Assign giftees" : "Only creator can assign giftees"}
                </Button>




            </CardContent>
        </Card>
    );
}