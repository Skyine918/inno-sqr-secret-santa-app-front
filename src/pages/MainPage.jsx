import * as React from 'react';
import {styled, useTheme} from '@mui/material/styles';
import Box from '@mui/material/Box';
import Drawer from '@mui/material/Drawer';
import CssBaseline from '@mui/material/CssBaseline';
import MuiAppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import List from '@mui/material/List';
import Typography from '@mui/material/Typography';
import Divider from '@mui/material/Divider';
import IconButton from '@mui/material/IconButton';
import MenuIcon from '@mui/icons-material/Menu';
import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';
import ListItem from '@mui/material/ListItem';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import GroupsIcon from '@mui/icons-material/Groups';
import InboxIcon from '@mui/icons-material/MoveToInbox';
import MailIcon from '@mui/icons-material/Mail';
import {Link} from "react-router-dom";
import Tooltip from "@mui/material/Tooltip";
import Avatar from "@mui/material/Avatar";
import Menu from "@mui/material/Menu";
import MenuItem from "@mui/material/MenuItem";
import {getAuth} from "@firebase/auth";
import {useAuthState} from "react-firebase-hooks/auth";
import Button from "@mui/material/Button";
import FirebaseGoogleAuth2Login from "./FirebaseGoogleAuth2Login";

const drawerWidth = 240;

const Main = styled('main', {shouldForwardProp: (prop) => prop !== 'open'})(
    ({theme, open}) => ({
        flexGrow: 1,
        padding: theme.spacing(3),
        transition: theme.transitions.create('margin', {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.leavingScreen,
        }),
        marginLeft: `-${drawerWidth}px`,
        ...(open && {
            transition: theme.transitions.create('margin', {
                easing: theme.transitions.easing.easeOut,
                duration: theme.transitions.duration.enteringScreen,
            }),
            marginLeft: 0,
        }),
    }),
);

const AppBar = styled(MuiAppBar, {
    shouldForwardProp: (prop) => prop !== 'open',
})(({theme, open}) => ({
    transition: theme.transitions.create(['margin', 'width'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
    }),
    ...(open && {
        width: `calc(100% - ${drawerWidth}px)`,
        marginLeft: `${drawerWidth}px`,
        transition: theme.transitions.create(['margin', 'width'], {
            easing: theme.transitions.easing.easeOut,
            duration: theme.transitions.duration.enteringScreen,
        }),
    }),
}));

const DrawerHeader = styled('div')(({theme}) => ({
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
    justifyContent: 'flex-end',
}));

export default function MainPage(props) {
    const theme = useTheme();
    const [open, setOpen] = React.useState(true);
    const auth = getAuth();
    const [user, loading, error] = useAuthState(auth);

    const handleDrawerOpen = () => {
        setOpen(true);
    };

    const handleDrawerClose = () => {
        setOpen(false);
    };

    const [anchorElNav, setAnchorElNav] = React.useState(null);
    const [anchorElUser, setAnchorElUser] = React.useState(null);

    const handleOpenNavMenu = (event) => {
        setAnchorElNav(event.currentTarget);
    };
    const handleOpenUserMenu = (event) => {
        setAnchorElUser(event.currentTarget);
    };

    const handleCloseNavMenu = () => {
        setAnchorElNav(null);
    };

    const handleCloseUserMenu = () => {
        setAnchorElUser(null);
    };

    if (!user) {
        return <FirebaseGoogleAuth2Login/>
    }

    return (
        <Box sx={{display: 'flex'}}>
            <CssBaseline/>
            <AppBar position="fixed" open={open}>
                <Toolbar>
                    <IconButton
                        data-testid="auth-icon-button"
                        color="inherit"
                        aria-label="open drawer"
                        onClick={handleDrawerOpen}
                        edge="start"
                        sx={{mr: 2, ...(open && {display: 'none'})}}
                    >
                        <MenuIcon/>
                    </IconButton>
                    <Typography
                        variant="h6"
                        noWrap
                        component="div"
                        sx={{mr: 2, display: {xs: 'none', md: 'flex'}}}
                    >
                        <Link id="nav-logo" to="/"
                              style={{display: "flex", color: "aliceblue", textDecoration: "none"}}>
                            <img style={{height: "2em", width: "2em", borderRadius: "50%"}} src="red-santa-hat-flat.png"
                                 alt=""/>
                            <span style={{lineHeight: "2em", marginLeft: "0.5em"}}>Secret Santa Web</span>
                        </Link>
                    </Typography>

                    <Box sx={{flexGrow: 1, display: {xs: 'none', md: 'flex', alignItems: 'center'}}}>

                        <Link style={{textDecoration: 'none', color: "white", textDecorationLine: true}} to="/"><Button
                            onClick={handleCloseNavMenu} sx={{my: 2, color: 'white', display: 'block', margin: 0}}>
                            SOME PAGE
                        </Button></Link>
                        <Link style={{textDecoration: 'none', color: "white", textDecorationLine: true}} to="/about-us">
                            <Button onClick={handleCloseNavMenu}
                                    sx={{my: 2, color: 'white', display: 'block', margin: 0}}>ABOUT US</Button>
                        </Link>
                    </Box>

                    {user ? (
                        <Button
                            color="secondary"
                            key="1"
                            onClick={handleCloseNavMenu}
                            variant="contained"
                            sx={{marginRight: '12px'}}
                        >
                            {!!user.displayName ? user.displayName : user.email}
                        </Button>
                    ) : (
                        ''
                    )}

                    <Box sx={{flexGrow: 0}}>
                        <Tooltip title="Open settings">
                            <IconButton onClick={handleOpenUserMenu} sx={{p: 0}}>
                                <Avatar alt={user ? user.displayName : "username"} src={user ? user.photoURL : ""}/>
                            </IconButton>
                        </Tooltip>
                        <Menu
                            sx={{mt: '45px'}}
                            id="menu-appbar"
                            anchorEl={anchorElUser}
                            anchorOrigin={{
                                vertical: 'top',
                                horizontal: 'right',
                            }}
                            keepMounted
                            transformOrigin={{
                                vertical: 'top',
                                horizontal: 'right',
                            }}
                            open={Boolean(anchorElUser)}
                            onClose={handleCloseUserMenu}
                        >
                            {user
                                ? <MenuItem key="sign-out" onClick={() => auth.signOut()}>
                                    <Typography textAlign="center">Logout</Typography>
                                </MenuItem>
                                : <MenuItem key="sign-in" onClick={() => auth.signOut()}>
                                    <Typography textAlign="center">
                                        <Link className="no-style-link" to="/sign-in">
                                            Sign In
                                        </Link>
                                    </Typography>
                                </MenuItem>}

                        </Menu>
                    </Box>
                </Toolbar>
            </AppBar>
            <Drawer
                sx={{
                    width: drawerWidth,
                    flexShrink: 0,
                    '& .MuiDrawer-paper': {
                        width: drawerWidth,
                        boxSizing: 'border-box',
                    },
                }}
                variant="persistent"
                anchor="left"
                open={open}
            >
                <DrawerHeader>
                    <IconButton onClick={handleDrawerClose}>
                        {theme.direction === 'ltr' ? <ChevronLeftIcon/> : <ChevronRightIcon/>}
                    </IconButton>
                </DrawerHeader>
                <Divider/>
                <List>
                    <Link to="/groups"><ListItem button>
                        <ListItemIcon>
                            <GroupsIcon/>
                        </ListItemIcon>
                        <ListItemText primary="My Groups"/>
                    </ListItem></Link>
                </List>
                <Divider/>
                <List>
                    {['All mail', 'Trash', 'Spam'].map((text, index) => (
                        <ListItem button key={text}>
                            <ListItemIcon>
                                {index % 2 === 0 ? <InboxIcon/> : <MailIcon/>}
                            </ListItemIcon>
                            <ListItemText primary={text}/>
                        </ListItem>
                    ))}
                </List>
            </Drawer>
            <Main open={open} style={{minHeight: "100vh", backgroundColor: "white"}}>
                <DrawerHeader/>
                {props.children}
            </Main>
        </Box>
    );
};