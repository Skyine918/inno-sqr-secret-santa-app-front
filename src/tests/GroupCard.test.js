import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Button from "@mui/material/Button";

import {render, cleanup, waitFor} from "@testing-library/react";
import "@testing-library/jest-dom"
import GroupCard from "../pages/GroupCard";
import {createMuiTheme, createTheme, ThemeProvider} from "@mui/material";
import {BrowserRouter} from "react-router-dom";
import MainPage from "../pages/MainPage";
import AppRouter from "../ui/AppRouter";
import {useAuthState} from "react-firebase-hooks/auth";
import CircularProgress from "@mui/material/CircularProgress";
import {initializeApp} from "firebase/app";
import {firebaseConfig} from "../globals";
import {getAuth} from "@firebase/auth";
import {MemoryRouter} from "react-router";
import {fireEvent, getByTestId} from "@testing-library/dom";

afterEach(cleanup)

it('matches snapshot', () => {
    const tree = renderer.create(<GroupCard
        tests={true}
        onAction={() => {}}
        user={{email: "test@santa.com"}}
        name="Test group"
        wishlist="Эчпочмак домашний"
        creator="test-creator@santa.com"
        assignee_wishlist={null}
        assignee_email={null}
        event_id={123}
        totalInvitedUsers={10}
        totalAcceptedUsers={10}
        location="Innopolis 106"
        date="01-01-2023"
        status="accepted"
    />).toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders base page', async () => {
    initializeApp(firebaseConfig);

    const theme = createTheme({
        palette: {
            primary: {
                main: '#f76C6C'
            },
            secondary: {
                main: '#f0f0f0'
            }
        }
    });

    const div = document.createElement('div');

    const { getByTestId, asFragment, container, findByText, findByDisplayValue } = render(
        <ThemeProvider theme={theme}>
            <MemoryRouter initialEntries={['/', '/groups']}>
                <MainPage>
                    <AppRouter user={{email: "test@santa.com"}}/>
                </MainPage>
            </MemoryRouter>
        </ThemeProvider>,
        div
    );
    const loginPasswordField = await waitFor(() => getByTestId('login-password-input'));
    const loginEmailField = await waitFor(() => getByTestId('login-email-input'));
    const inputEmail = container.querySelector("#santa-auth-email")
    const inputPassword = container.querySelector("#santa-auth-password")

    fireEvent.change(inputEmail, {target: {value: 'test1@santa.com'}});
    await (waitFor(() => findByDisplayValue("test1@santa.com"),{timeout:3000, interval: 1000}));
    fireEvent.change(inputPassword, {target: {value: 'qwertu'}});
    await (waitFor(() => findByDisplayValue("qwertu"),{timeout:3000, interval: 1000}));
    fireEvent.click(await getByTestId("login-button"));

    expect(inputEmail.value).toBe('test1@santa.com')

    const helpText = await (waitFor(() => findByText("Invalid Email Or Password"),{timeout:3000, interval: 1000}));
    console.log(helpText)

});

