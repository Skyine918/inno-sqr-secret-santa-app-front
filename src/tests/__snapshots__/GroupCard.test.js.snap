// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders base page 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
>
  <div
    style={
      Object {
        "alignItems": "center",
        "background": "white",
        "display": "flex",
        "flexDirection": "column",
        "marginTop": "20vh",
        "padding": "2em 1em 1em 1em",
      }
    }
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-m3aefh-MuiAvatar-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LockOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Sign In
    </h1>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-z9gj7g-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-13qr752-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="santa-auth-email"
        id="santa-auth-email-label"
      >
        Email
        <span
          aria-hidden={true}
          className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-ws5skk-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-describedby="santa-auth-email-helper-text"
          aria-invalid={false}
          autoComplete="santa-auth-email"
          autoFocus={true}
          className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="santa-auth-email"
          name="Email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1g8cwq9"
          >
            <span>
              Email
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-sizeSmall MuiFormHelperText-contained Mui-required css-k4qjio-MuiFormHelperText-root"
        id="santa-auth-email-helper-text"
      >
        Your email
      </p>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-z9gj7g-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-13qr752-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="santa-auth-password"
        id="santa-auth-password-label"
      >
        Password
        <span
          aria-hidden={true}
          className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-ws5skk-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="santa-auth-password"
          autoFocus={true}
          className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="santa-auth-password"
          name="Password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1g8cwq9"
          >
            <span>
              Password
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-ke5b6m-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      id="sign-in-button"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Sign in
    </button>
    <p
      className="MuiTypography-root MuiTypography-small css-37jdci-MuiTypography-root"
      style={
        Object {
          "color": "black",
          "margin": "0.5em 0",
        }
      }
    >
      OR
    </p>
    <button
      className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-18v26h3-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "background": "white",
          "marginTop": "10px",
          "paddingRight": "60px",
          "textTransform": "none",
        }
      }
      tabIndex={0}
      type="button"
    >
      <img
        alt="google"
        src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBRF__vDRer9N3lzmW-FJTnaiCi1Vd7TvcHrdcjzU28RHD2kcpRVdZIQhLvZaksbBPpak&usqp=CAU"
        style={
          Object {
            "height": "20px",
            "marginRight": "60px",
            "width": "20px",
          }
        }
      />
      Log in with Google
    </button>
  </div>
</main>
`;
